pykube/config.py

```diff
@@ -115,12 +115,17 @@ def users(self):
         Returns known users by exposing as a read-only property.
         """
         if not hasattr(self, "_users"):
-            us = {}
-            for ur in self.doc["users"]:
-                us[ur["name"]] = u = copy.deepcopy(ur["user"])
-                BytesOrFile.maybe_set(u, "client-certificate")
-                BytesOrFile.maybe_set(u, "client-key")
+            us = {
+                "default": {}
+            }
+
+            if "users" in self.doc:
```

Comment:
This looks necessary, but what is the purpose of `default`? It doesn't seem like anything is using it in your changes or in pykube generally. Wouldn't an empty dictionary be enough.

As I think about this some more, there appears there may be a bug in relation to this. If there is no users in the config and `KubeConfig.user` requires there to be a user for the current context, won't there still be a failure (unless I did somehow miss that `default` user is being used somehow. Perhaps we need better handling in `user` to allow it to return `None`.